src/pages/home/report/ContextMenu/ContextMenuActions.js

```diff
@@ -96,20 +97,28 @@ export default [
         // the `text` and `icon`
         onPress: (closePopover, {reportAction, selection}) => {
             const message = _.last(lodashGet(reportAction, 'message', [{}]));
-            const html = lodashGet(message, 'html', '');
-
-            const parser = new ExpensiMark();
-            const reportMarkdown = parser.htmlToMarkdown(html);
-
-            const text = selection || reportMarkdown;
+            const messageHtml = lodashGet(message, 'html', '');
 
             const isAttachment = _.has(reportAction, 'isAttachment')
                 ? reportAction.isAttachment
                 : ReportUtils.isReportMessageAttachment(message);
             if (!isAttachment) {
-                Clipboard.setString(text);
+                // This error indicates new code that uses the selection object incorrectly.
+                if (selection && (selection.text || !selection.html)) {
+                    throw new Error('Selection has to either exclusively bear html or be null.');
+                }
+                const resolvedSelection = selection || SelectionScraper.prepareSelection(messageHtml);
+                if (!resolvedSelection) {
+                    return;
+                }
+                const parser = new ExpensiMark();
+                if (!Clipboard.canSetHtml()) {
+                    Clipboard.setString(parser.htmlToMarkdown(resolvedSelection.html));
+                    return;
```

Comment:
Unfortunately, you can't early return here. Don't forget closePopover condition.